lib/services/apple-portal/apple-portal-session-service.ts

```diff
@@ -79,6 +75,72 @@ export class ApplePortalSessionService implements IApplePortalSessionService {
 		return webSessionCookie;
 	}
 
+	private async login(credentials: ICredentials, opts?: IAppleCreateUserSessionOptions): Promise<IAppleLoginResult> {
+		const result = {
+			scnt: <string>null,
+			xAppleIdSessionId: <string>null,
+			isTwoFactorAuthenticationEnabled: false,
+			areCredentialsValid: true
+		};
+
+		if (opts && opts.sessionBase64) {
+			const decodedSession = Buffer.from(opts.sessionBase64, "base64").toString("utf8");
+
+			this.$applePortalCookieService.updateUserSessionCookie([decodedSession]);
+
+			result.isTwoFactorAuthenticationEnabled = decodedSession.indexOf("DES") > -1;
+		} else {
+			try {
+				await this.loginCore(credentials);
+			} catch (err) {
+				const statusCode = err && err.response && err.response.statusCode;
+				result.areCredentialsValid = statusCode !== 401 && statusCode !== 403;
+				result.isTwoFactorAuthenticationEnabled = statusCode === 409;
+				if (result.isTwoFactorAuthenticationEnabled && opts && !opts.applicationSpecificPassword) {
+					this.$errors.failWithoutHelp(`Your account has two-factor authentication enabled but --appleApplicationSpecificPassword option is not provided.
+To generate an application-specific password, please go to https://appleid.apple.com/account/manage.
+This password will be used for the iTunes Transporter, which is used to upload your application.`);
+				}
+
+				if (result.isTwoFactorAuthenticationEnabled && opts && opts.ensureConsoleIsInteractive && !isInteractive()) {
+					this.$errors.failWithoutHelp(`Your account has two-factor authentication enabled, but your console is not interactive.
```

Comment:
We could mention the `appleSessionBase64` option here.

Reply:
I was wondering the same and decided that mention `tns publish ios --help` command and describing `--appleSessionBase64` option in help, should be enough.